addons/sale_timesheet/tests/test_sale_service.py

```diff
@@ -92,9 +92,9 @@ def test_sale_service(self):
         # delete timesheets before deleting the task, so as to trigger the error
         # about linked sales order lines and not the one about linked timesheets
         task.timesheet_ids.unlink()
-        # not possible to delete a task linked to a SOL
-        with self.assertRaises(ValidationError):
-            task.unlink()
+        # unlink automatically task from the SOL when deleting the task
+        task.unlink()
+        self.assertEqual(self.sale_order.tasks_count, 1, "Deleteing the task its should automatically unlink the task from SOL.")
```

Comment:
I mean when a SOL generates a task, the task generated is stored in `task_id` field. So, I think in this case, the `sale_order_line` variable contained the SOL that it generates the task. So, normally the task generated is added in the `task_id` field of this SOL. Can you check if the task_id field is False when we remove the task. 🙂 
```suggestion
        self.assertFalse(sale_order_line.task_id, "Deleting the task its should automatically unlink the task from SOL.")
```